### ISSUE TITLE
[REVIEW]: J2suscep: Calculation of magnetic exchange coupling and temperature dependence of magnetic susceptibility

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2838]
**Submitting author:** @Swetanshu (<a href="http://orcid.org/0000-0002-0626-0626">Swetanshu Tandon</a>)
**Repository:** <a href="https://github.com/WatsonGroupTCD/J2suscep" target ="_blank">https://github.com/WatsonGroupTCD/J2suscep</a>
**Version:** 1.0
**Editor:** @dpsanders
**Reviewer:** @mturiansky, @mfherbst
**Archive:** <a href="https://doi.org/10.5281/zenodo.5024912" target="_blank">10.5281/zenodo.5024912</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2/status.svg)](https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2"><img src="https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2/status.svg)](https://joss.theoj.org/papers/9d61553409a3cb055ecb4feb7feb5de2)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/WatsonGroupTCD/J2suscep" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mturiansky & @mfherbst, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dpsanders know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/WatsonGroupTCD/J2suscep">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Swetanshu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/WatsonGroupTCD/J2suscep">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Swetanshu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mturiansky, @mfherbst it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/jcc.23234 is OK
- 10.1021/ic900992r is OK
- 10.1039/B902745A is OK
- 10.1007/128_2015_645 is OK
- 10.1016/j.ccr.2008.05.014 is OK
- 10.1021/ic502651w is OK
- 10.1016/j.cpc.2010.09.007 is OK
- 10.1039/D0SE00701C is OK
- 10.1039/D0DT01404D is OK
- 10.1002/chem.201405679 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by mturiansky** [comment]
@dpsanders a few quick questions: with regards to the "software paper" section, how strict is JOSS? Does there need to be an explicit section labeled "summary" or "state of the field". I feel that overall the paper covers these two topics, but not directly in their own section.

How about with writing quality? The paper gets its point across, but there are grammatical errors (run-on sentences, missing commas) that could be addressed.

I would say at this point that I have completed my review. I have opened several issues, and I would recommend major revisions. There is no installation instructions, so without major effort on my part, I cannot compile, run, or verify the output of the code. Also, there are no unit tests, which will likely require a not-insignificant effort to develop.

**Comment by mfherbst** [comment]
@dpsanders I am now also done with my review and would recommend major revisions. I opened up various issues with respect to my points of critique or added some comments to the earlier ones of @mturiansky.

Similar to @mturiansky I wonder a little about the Software paper section. I found the summary and statement of need addressed
to a satisfactory point, albeit not in particularly labelled sections. In contrast I am not so sure about the description of the state of the art. While I am not an expert in this particular subfield of quantum chemistry, it seems to me the authors solve a common enough problem, such that it could be worth expanding on the relationship of their code compared to other solutions in the field. Since JOSS is more centred around the code rather than the scientific methods, I am not sure how strict JOSS is in this respect, though.

Another aspect is the quality of the code itself. From the snippets I read the code makes use of some features that might make it difficult to maintain in the long run (e.g. global variables). How much is this a concern for JOSS? Is it about the functionality of the present stage of the code or is it also about the long-term maintainability and the value a project adds to the open-source community?

Despite clearly involving some effort I think the code at its present stage could benefit a lot from taking the time to come up with a better structure, such that e.g. code duplication or global variables are avoided. This would not only make it more accessible for others, but also easier to maintain and test.

**Comment by whedon** [comment]
:wave: @mturiansky, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @mfherbst, please update us on how your review is going.

**Comment by dpsanders** [comment]
👋 Hi @mturiansky and @mfherbst: Many thanks for your comments and sorry for the delay in getting to them (and for the automated reminders!). I hope to get to look through them carefully today or tomorrow.

**Comment by dpsanders** [comment]
👋 Hi @mturiansky and @mfherbst: Many thanks for your detailed and perceptive comments and issues on the repo. My apologies for the long delay in getting to this.

I agree with all your points (modulo a couple of clarifying remarks in the issues). @swetanshu: I would ask you to please address the points raised in the issues on the repo, which will make for a more useful and reusable piece of software. I will respond to particular points below.

**Comment by dpsanders** [comment]
@mturiansky: 

> with regards to the "software paper" section, how strict is JOSS? Does there need to be an explicit section labeled "summary" or "state of the field". I feel that overall the paper covers these two topics, but not directly in their own section.

HOSS has no explicit requirements about this, but I agree that that would improve the structure of the paper.

> How about with writing quality? The paper gets its point across, but there are grammatical errors (run-on sentences, missing commas) that could be addressed.

The text should be readable; if it isn't, that is unacceptable. I am happy to help with minor grammar and punctuation issues once the paper is ready.

> There is no installation instructions, so without major effort on my part, I cannot compile, run, or verify the output of the code. Also, there are no unit tests, which will likely require a not-insignificant effort to develop.

Installation instructions are a requirement of JOSS, since without them the software is, as you say, unusable. It is a requirement for their to be tests; the tests are not *required* to be automated, but I agree that this is a standard feature in modern software.

Thanks again for your comments!

**Comment by dpsanders** [comment]
@mfherbst: 


> I am not so sure about the description of the state of the art. While I am not an expert in this particular subfield of quantum chemistry, it seems to me the authors solve a common enough problem, such that it could be worth expanding on the relationship of their code compared to other solutions in the field. Since JOSS is more centred around the code rather than the scientific methods, I am not sure how strict JOSS is in this respect, though.

It is a requirement to describe the relation of the software with other common packages in the same area. Implicitly that requires specifying how this particular piece of software is different from the others; this is related to the Statement of need, which needs to say why *this* particular piece of software is required.

In particular, https://joss.readthedocs.io/en/latest/submitting.html states that "[the] software should be a significant contribution to the available open source software that either enables some new research challenges to be addressed or makes addressing research challenges significantly better (e.g., faster, easier, simpler)." It should thus be explicitly addressed in the paper in what way this is accomplished.

> Another aspect is the quality of the code itself. From the snippets I read the code makes use of some features that might make it difficult to maintain in the long run. Despite clearly involving some effort I think the code at its present stage could benefit a lot from taking the time to come up with a better structure, such that e.g. code duplication or global variables are avoided. This would not only make it more accessible for others, but also easier to maintain and test.

> How much is this a concern for JOSS? Is it about the functionality of the present stage of the code or is it also about the long-term maintainability and the value a project adds to the open-source community?

It is a requirement (see the above-cited page) that "JOSS requires that software should be feature-complete... and designed for maintainable extension... single-function packages are not acceptable."  

So I agree with you that the software should be written in a way that enables other people to understand its structure and maintain it. If you think the current structure of the code does not allow that then it needs to be modified for publication in JOSS.

Thanks again for your comments!

**Comment by Swetanshu** [comment]
Thank you @mturiansky and @mfherbst for the quick review, and @dpsanders for your response!

We have been working on updating the code in the past few days. I was not sure whether I should make any commits before @dpsanders gets back so I did not do that. I’ll start committing the updates that have been made. To summarise, the code has been restructured, and examples and documentation have been updated. We are currently working on implementing automated tests. I’ll get back to you as soon as that is done.

**Comment by Swetanshu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by Swetanshu** [comment]
Once again thank you @mturiansky, @mfherbst and @dpsanders  for your feedback. Apologies for taking this long to get back to you. We have tried to incorporate almost everything that was suggested. Briefly, the workflow has been streamlined and avoids code duplication, and the repository has been restructured for clarity. The wiki pages include installation and contribution instructions, and simpler examples. Automated tests have also been included to test the proper operation of the code. We have responded to each issue separately in the issue tracker to describe how we addressed each issue. I’ll close them if you are happy with the responses.
The paper has also been updated to remove grammatical errors. We summarise the code and provide an example under the heading ‘J2suscep’. We have not created a separate heading for ‘state of the field’ but we explicitly mention in the paper that the purpose served by ej_calc is not covered elsewhere, and we have included suscep to make the code self-contained.
I look forward to hearing back from you!

**Comment by dpsanders** [comment]
👋 @mturiansky and @mfherbst: Could you please check the current version of the code and paper and update your checkboxes accordingly. Thanks!

**Comment by mfherbst** [comment]
Hi all ... short update from me. I'm going through a pretty busy period at the moment, so I'm not sure I'll get to doing the review within the next two weeks. Sorry about that.

**Comment by dpsanders** [comment]
👋 @mfherbst and @mturiansky: Any updates here? Thanks!

**Comment by mturiansky** [comment]
Sorry, I should have updated the thread. I checked off a few boxes above, closed some of the issues that were addressed, and opened a new minor issue. @dpsanders I tagged you in one of the issues with a minor question about JOSS's standards. Other than that the code looks to be closer to publishable quality.

**Comment by mfherbst** [comment]
I have an important deadline this Thursday. I should get to do the review the latest next week.

**Comment by mfherbst** [comment]
On my end I've just done a similar round of updates as @mturiansky. I ticked off some extra boxes and closed some issues, but I would recommend another round of polishing.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by Swetanshu** [comment]
Once again thank you @mturiansky, @mfherbst  and @dpsanders  for your feedback. We have tried to tidy up the code further and incorporate the suggested functionalities. We have responded to each issue separately in the issue tracker to describe how we addressed each issue. I’ll close them if you are happy with the responses. 
I look forward to hearing back from you.

**Comment by mturiansky** [comment]
@dpsanders, all of the issues that I have opened are now closed. I think once the remaining issues are addressed (assuming no new issues present themselves), then I'm okay with publication.

**Comment by mfherbst** [comment]
Same here. My issues were either addressed sufficiently or closed. There is still a typo I reported, but otherwise I'd recommend publication.

**Comment by Swetanshu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by Swetanshu** [comment]
Once again thank you @mturiansky, @mfherbst and @dpsanders for your feedback. Now, the usage information for the script is printed when –h tag is used or when invalid input is provided.
@mfherbst The typo in the paper has been removed.
We have responded to the issues separately in the issue tracker as well. I’ll close them if you are happy with the responses. 
@dpsanders please let us know if you are happy with the responses to the issues that you opened.
I look forward to hearing back from you.

**Comment by danielskatz** [comment]
👋 @dpsanders - It looks like everything is checked off from the reviewers - can you help me understand what this submission is waiting for at this point?

**Comment by dpsanders** [comment]
Apologies for the delay. I will be doing a final look tomorrow.

**Comment by dpsanders** [comment]
@Swetanshu: I have sent a PR for the JOSS paper with minor edits. 

I have also requested that the worked example be extended to show how to run Gaussian, get the output and pass it to your tools to make a complete worked example.

**Comment by Swetanshu** [comment]
@dpsanders Thanks a lot for this!
The worked example includes more details now. The edits you requested have been merged with the main branch.
Let us know if anything else is needed or if I should close all the issues that remain open.
Thanks again!

**Comment by Swetanshu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
@whdeon commands

**Comment by dpsanders** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by dpsanders** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/jcc.23234 is OK
- 10.1002/(SICI)1096-987X(199910)20:13<1391::AID-JCC6>3.0.CO;2-J is OK
- 10.1021/ic900992r is OK
- 10.1039/B902745A is OK
- 10.1007/128_2015_645 is OK
- 10.1016/j.ccr.2008.05.014 is OK
- 10.1021/ic502651w is OK
- 10.1016/j.cpc.2010.09.007 is OK
- 10.1039/D0SE00701C is OK
- 10.1039/D0DT01404D is OK
- 10.1021/ct2006506 is OK
- 10.1002/chem.201405679 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1002/0470862106.ia618 is INVALID because of 'https://doi.org/' prefix
```

**Comment by dpsanders** [comment]
@Swetanshu: Please fix the invalid DOI listed above. 
Also the paper by Rajeshkumar is missing the title.

**Comment by dpsanders** [comment]
👋 @Swetanshu: Once you have done that, please archive the repository at Zenodo or similar, making sure that the title and authors arer exactly the same as in the JOSS paper.

Then please report here the software version and the DOI of the repository.

Thanks!

**Comment by Swetanshu** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/jcc.23234 is OK
- 10.1002/0470862106.ia618 is OK
- 10.1002/(SICI)1096-987X(199910)20:13<1391::AID-JCC6>3.0.CO;2-J is OK
- 10.1021/ic900992r is OK
- 10.1039/B902745A is OK
- 10.1007/128_2015_645 is OK
- 10.1016/j.ccr.2008.05.014 is OK
- 10.1021/ic502651w is OK
- 10.1016/j.cpc.2010.09.007 is OK
- 10.1039/D0SE00701C is OK
- 10.1039/D0DT01404D is OK
- 10.1021/ct2006506 is OK
- 10.1002/chem.201405679 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Swetanshu** [comment]
@dpsanders The software has been archived on zenodo. The doi is 10.5281/zenodo.5024912
The version is 1.0.
Let me know if there is anything else that needs to be done from my end.
Also, can you let me know if I can close the issues that are still open?

Thanks again!

**Comment by dpsanders** [comment]
@Swetanshu: Please make sure that the title and author list on Zenodo exactly match the JOSS paper.

**Comment by dpsanders** [comment]
@whedon set 1.0 as version

**Comment by whedon** [comment]
OK. 1.0 is the version.

**Comment by dpsanders** [comment]
@whedon set 10.5281/zenodo.5024912 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5024912" target="_blank">10.5281/zenodo.5024912</a> is the archive.

**Comment by dpsanders** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02838/joss.02838/10.21105.joss.02838.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
@Swetanshu: The paper by Rajeshkumar is still missing a title in the References section.

**Comment by Swetanshu** [comment]
@dpsanders: I don't really understand this because I can see the title in the pdf file that you have generated. Can you please check it again?
![rajesh](https://user-images.githubusercontent.com/61150617/124552485-e87aea80-de33-11eb-849b-d221a51c33e1.png)

**Comment by dpsanders** [comment]
@Swetanshu: Apologies; I see it now. 
The metadata (title and authors) on Zenodo is still not correct, though.

**Comment by Swetanshu** [comment]
@dpsanders Sorry, but I am not really sure what you want me to change. Can you please explain a little about what needs to be done and how do I go about it. Thanks in advance!

**Comment by dpsanders** [comment]
@Swetanshu: The title of the Zenodo repository must be exactly the same as the title of the JOSS paper. And the author list must also be exactly the same.

**Comment by Swetanshu** [comment]
@dpsanders The title and the author list have been modified on Zenodo. Let me know if everything is in order now or if there is anything else that needs to be modified.
Thanks again!

**Comment by danielskatz** [comment]
👋 @dpsanders - it looks like this is ready for your final processing?

**Comment by dpsanders** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1002/jcc.23234 is OK
- 10.1002/0470862106.ia618 is OK
- 10.1002/(SICI)1096-987X(199910)20:13<1391::AID-JCC6>3.0.CO;2-J is OK
- 10.1021/ic900992r is OK
- 10.1039/B902745A is OK
- 10.1007/128_2015_645 is OK
- 10.1016/j.ccr.2008.05.014 is OK
- 10.1021/ic502651w is OK
- 10.1016/j.cpc.2010.09.007 is OK
- 10.1039/D0SE00701C is OK
- 10.1039/D0DT01404D is OK
- 10.1021/ct2006506 is OK
- 10.1002/chem.201405679 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2465

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2465, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2466
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02838](https://doi.org/10.21105/joss.02838)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to  @Swetanshu (Swetanshu Tandon) and co-authors!!

And thanks to @mturiansky and @mfherbst for reviewing, and @dpsanders for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02838/status.svg)](https://doi.org/10.21105/joss.02838)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02838">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02838/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02838/status.svg
   :target: https://doi.org/10.21105/joss.02838
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02838/status.svg)](https://doi.org/10.21105/joss.02838)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
